phoenix-core/src/main/java/org/apache/phoenix/mapreduce/ViewTTLDeleteJobMapper.java

```diff
@@ -0,0 +1,128 @@
+            } else {
+                LOGGER.info("Using default tracker class ");
+            }
+            this.multiViewJobStatusTracker = (MultiViewJobStatusTracker) defaultViewDeletionTrackerClass.newInstance();
+        } catch (Exception e) {
+            LOGGER.error("Getting exception While initializing initMultiViewJobStatusTracker with error message");
+            LOGGER.error("stack trace" + e.getStackTrace().toString());
+            throw e;
+        }
+    }
+
+    @Override
+    protected void map(NullWritable key, ViewInfoTracker value, Context context) throws IOException  {
+        try {
+            final Configuration config = context.getConfiguration();
+
+            if (this.multiViewJobStatusTracker == null) {
+                initMultiViewJobStatusTracker(config);
+            }
+
+            LOGGER.debug(String.format("Deleting from view %s, TenantID %s, and TTL value: %d",
+                    value.getViewName(), value.getTenantId(), value.getViewTtl()));
+
+            try (PhoenixConnection connection = (PhoenixConnection) ConnectionUtil.getInputConnection(config)) {
+                if (value.getTenantId() != null && !value.getTenantId().equals("NULL")) {
+                    try (PhoenixConnection tenantConnection = (PhoenixConnection) PhoenixMultiInputUtil.
+                            buildTenantConnection(connection.getURL(), value.getTenantId())) {
+                        deletingExpiredRows(tenantConnection, value, config);
+                    }
+                } else {
+                    deletingExpiredRows(connection, value, config);
+                }
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Mapper got an exception while deleting expired rows : " + e.getMessage() );
+            throw new IOException(e.getMessage(), e.getCause());
+        } catch (Exception e) {
+            LOGGER.error("Getting IOException while running View TTL Deletion Job mapper with error : "
+                    + e.getMessage());
+            throw new IOException(e.getMessage(), e.getCause());
+        }
+    }
+
+    private void deletingExpiredRows(PhoenixConnection connection, ViewInfoTracker view, Configuration config)
+            throws SQLException {
+
+        String deleteIfExpiredStatement = "DELETE FROM " + view.getViewName() +
+                " WHERE TO_NUMBER(NOW()) - TO_NUMBER(PHOENIX_ROW_TIMESTAMP()) > " + view.getViewTtl();
+
+        try {
+            this.multiViewJobStatusTracker.updateJobStatus(view, 0,
+                    ViewInfoJobState.PREP.getValue(), config, 0);
+
+            long startTime = System.currentTimeMillis();
+            this.multiViewJobStatusTracker.updateJobStatus(view, 0,
+                    ViewInfoJobState.RUNNING.getValue(), config, 0 );
+            connection.setAutoCommit(true);
+            int numberOfDeletedRows = connection.createStatement().executeUpdate(deleteIfExpiredStatement);
```

Comment:
create statement inside try-with-resources. We have to very careful not to leak any connections/statements, etc. from our own framework. Can you please ensure that throughout? I may have missed some.